history.py

```diff
@@ -0,0 +1,35 @@
+from catalog import models as cmod
+
+class LastFiveMiddleware:
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        # Code to be executed for each request before
+        # the view (and later middleware) are called.
+        id_list = request.session.get('id_list', [])
+        try:
+            id_list = request.session['id_list']
+        except:
+            id_list = []
+
+        product_list = []
+        if len(id_list) != 0:
+            for pid in id_list:
+                product_list.append(cmod.Product.objects.get(id = pid ))
+
+        request.last_five = product_list
+        print("hola")
```

Comment:
The testing could be a bit more helpful and print a value or something in order to assist with debugging.

Reply:
Yeah, I dealt with that more in my detail.py file